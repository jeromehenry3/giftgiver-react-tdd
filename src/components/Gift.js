import React, { Component } from 'react';
import { Form, Button } from 'react-bootstrap';

class Gift extends Component {
    state = {
        person: '',
        present: '',
    }

    render() {
        return (
            <div>
                <Form>
                    <Form.Group>
                        <Form.Label>Personne</Form.Label>
                        <Form.Control
                            onChange={evt => this.setState({ person: evt.target.value })}
                            className='input-person'
                        />
                    </Form.Group>
                    <Form.Group>
                        <Form.Label>Cadeau</Form.Label>
                        <Form.Control
                            onChange={evt => this.setState({ present: evt.target.value })}
                            className='input-present'
                        />
                    </Form.Group>
                </Form>
                <Button
                    className='btn-remove'
                    onClick={() => this.props.removeGift(this.props.gift.id)}
                >Retirer le cadeau</Button>
            </div>
        );
    }
}

export default Gift;