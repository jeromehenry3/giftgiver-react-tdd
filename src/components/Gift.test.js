import React from 'react';
import { shallow } from 'enzyme';
import Gift from './Gift';

describe('Gift', () => {
    const gift = shallow(<Gift />);

    it('renders properly', () => {
        expect(gift).toMatchSnapshot();
    });

    it('initializes a person and present in `state`', () => {
        expect(gift.state()).toEqual({ person: '', present: '' });
    })

    describe('when typing into the person input', () => {
        const person = 'Robert'

        beforeEach(() => {
            gift.find('.input-person').simulate('change', { target: { value: person } });
        });

        it('updates the person in the `state`', () => {
            expect(gift.state().person).toEqual(person);
        });
    });

    describe('when typing into the present input', () => {
        const present = 'Un livre de Spinoza';

        beforeEach(() => {
            gift.find('.input-present').simulate('change', { target: { value: present } });
        })
        it('updates the present in the `state`', () => {
            expect(gift.state().present).toEqual(present);
        })
    })
})
