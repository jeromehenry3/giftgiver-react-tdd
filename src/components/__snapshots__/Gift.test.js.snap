// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift renders properly 1`] = `
"<div>
  <Form inline={false}>
    <FormGroup>
      <FormLabel column={false} srOnly={false}>
        Personne
      </FormLabel>
      <FormControl onChange={[Function: onChange]} className=\\"input-person\\" />
    </FormGroup>
    <FormGroup>
      <FormLabel column={false} srOnly={false}>
        Cadeau
      </FormLabel>
      <FormControl onChange={[Function: onChange]} className=\\"input-present\\" />
    </FormGroup>
  </Form>
  <Button className=\\"btn-remove\\" onClick={[Function: onClick]} variant=\\"primary\\" active={false} disabled={false} type=\\"button\\">
    Retirer le cadeau
  </Button>
</div>"
`;
